core/src/test/java/com/linecorp/armeria/common/RequestContextTest.java

```diff
@@ -176,6 +178,21 @@ public void makeContextAwareChannelFutureListener() {
     }
 
     @Test
+    public void makeContextAwareCompletableFuture() throws Exception {
+        RequestContext context = createContext();
+        CompletableFuture<String> future = context.makeContextAware(
+                CompletableFuture.supplyAsync(() -> "success", ImmediateEventExecutor.INSTANCE));
```

Comment:
Hmm.. `assertBlahBlah`s are not work in another thread, because they are depend on throwing Errors. I'll update tests with real `Executor` and assertion on another thread.